docs/source/examples/word-count.rst

```diff
@@ -0,0 +1,247 @@
+
+Start the ``distributed`` scheduler and workers on the cluster.
+
+Code example
+------------
+
+Import ``distributed``, ``hdfs3``, and other standard libraries used in this example:
+
+.. code-block:: python
+
+   >>> import hdfs3
+   >>> from collections import defaultdict, Counter
+   >>> from distributed import Executor, progress
+
+Initalize a connection to HDFS, replacing ``NAMENODE_HOSTNAME`` and
+``NAMENODE_PORT`` with the hostname and port (default: 8020) of the HDFS
+namenode.
+
+.. code-block:: python
+
+   >>> hdfs = hdfs3.HDFileSystem('NAMENODE_HOSTNAME', port=NAMENODE_PORT)
+
+Initalize a connection to the ``distributed`` executor, replacing
+``EXECUTOR_IP`` and ``EXECUTOR_PORT`` with the hostname and port of the
+``distributed`` scheduler.
+
+.. code-block:: python
+
+   >>> e = Executor('EXECUTOR_IP:EXECUTOR_PORT')
+
+Generate a list of filenames from the text data in HDFS:
+
+.. code-block:: python
+
+   >>> filenames = hdfs.glob('/tmp/enron/*/*')
+   >>> print(filenames[:5])
+
+   ['/tmp/enron/edrm-enron-v2_nemec-g_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_ring-r_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_bailey-s_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_fischer-m_xml.zip/merged.txt',
+   '/tmp/enron/edrm-enron-v2_geaccone-t_xml.zip/merged.txt']
+
+Print the first 1024 bytes of the first text file:
+
+.. code-block:: python
+
+   >>> print(hdfs.head(filenames[0]))
+
+   b'Date: Wed, 29 Nov 2000 09:33:00 -0800 (PST)\r\nFrom: Xochitl-Alexis Velasc
+   o\r\nTo: Mark Knippa, Mike D Smith, Gerald Nemec, Dave S Laipple, Bo Barnwel
+   l\r\nCc: Melissa Jones, Iris Waser, Pat Radford, Bonnie Shumaker\r\nSubject:
+    Finalize ECS/EES Master Agreement\r\nX-SDOC: 161476\r\nX-ZLID: zl-edrm-enro
+   n-v2-nemec-g-2802.eml\r\n\r\nPlease plan to attend a meeting to finalize the
+    ECS/EES  Master Agreement \r\ntomorrow 11/30/00 at 1:30 pm CST.\r\n\r\nI wi
+   ll email everyone tomorrow with location.\r\n\r\nDave-I will also email you 
+   the call in number tomorrow.\r\n\r\nThanks\r\nXochitl\r\n\r\n***********\r\n
+   EDRM Enron Email Data Set has been produced in EML, PST and NSF format by ZL
+    Technologies, Inc. This Data Set is licensed under a Creative Commons Attri
+   bution 3.0 United States License <http://creativecommons.org/licenses/by/3.0
+   /us/> . To provide attribution, please cite to "ZL Technologies, Inc. (http:
+   //www.zlti.com)."\r\n***********\r\nDate: Wed, 29 Nov 2000 09:40:00 -0800 (P
+   ST)\r\nFrom: Jill T Zivley\r\nTo: Robert Cook, Robert Crockett, John Handley
+   , Shawna'
+
+Create a function to count words in each file:
+
+.. code-block:: python
+
+   >>> def count_words(fn):
+   ...     word_counts = defaultdict(int)
+   ...     with hdfs.open(fn) as f:
+   ...         for line in f:
+   ...             for word in line.split():
+   ...                 word_counts[word] += 1
+   ...     return word_counts
+
+Count the number of words in the first text file locally:
```

Comment:
This might deserve some prose saying why we would do this.  "Before we scale out to the full cluster, lets test our function on a single file using our local computer...